web/availabilitycounter_test.go

```diff
@@ -0,0 +1,52 @@
+package web
+
+import (
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/signalfx/golib/timekeeper/timekeepertest"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+type limitStub struct {
+	fastRequestLimit time.Duration
+}
+
+func (l limitStub) Get() time.Duration {
+	return l.fastRequestLimit
+}
+
+func TestAvailabilityCounter(t *testing.T) {
+	starttime := time.Date(1981, time.March, 19, 6, 0, 0, 0, time.UTC)
+	fastRequestLimit := 100 * time.Millisecond
+	limitStub := limitStub{
+		fastRequestLimit: fastRequestLimit,
+	}
+
+	Convey("when setup", t, func() {
+		timeStub := timekeepertest.NewStubClock(starttime)
+		counterHandler := AvailabilityCounterHandler{
+			fastRequestLimitDuration: limitStub,
+			totalFast:                0,
+			totalSlow:                0,
+			timeKeeper:               timeStub,
+		}
+		ctx := context.Background()
+		Convey("When request is slow", func() {
```

Comment:
The Convey together should read as a sentence.  So make this one "slow requests should count to the slow bucket" so that the total sentence becomes "When setup, slow requests should count to the slow bucket"